### DISCUSSION TITLE
Get values from related fields

### DISCUSSION THREAD

**Discussion by SaschaValentin** [Budibase/budibase#12188] [Help] [answered]
I have a table with news which is related to users table (author of a news entry is one user).

In the users table besides the firstname and lastname there is also a field avatar with an image.

When I´m building a screen for the news table, how can I retrieve the values of the linked users?

![ScreenShot 038 Budibase – Mozilla Firefox](https://github.com/Budibase/budibase/assets/48714823/714ef233-e64c-4d12-96fb-e9aff0d5fd47)

The red marked values should be different users, but it shows the currents users values. 

What I did: 

1. Data provider for the news table
2. In this provider there is a repeater for the table.
3. I guess my thinking fault is in this step: To get the values of users avatar I put another data provider pointing to data FIELDS > Current User.Avatar
![ScreenShot 039 Budibase – Mozilla Firefox](https://github.com/Budibase/budibase/assets/48714823/dcdf32bd-ed1b-4c70-b3e2-a3918fc6e09d)

4. Then I put another repeater and in that an image binding to {{ Image  Repeater.Avatar.url }}.

Hope there is someone who can explain what I did wrong and how to correct it.

**Comment by SaschaValentin** [comment]
I´ve solved the issue with the username myself. Just accessed the values of firstname and lastname from the users object with javascript: 

**return userobject[0]["firstName"]+" "+userobject[0]["lastName"];**

![ScreenShot 040 Budibase – Mozilla Firefox](https://github.com/Budibase/budibase/assets/48714823/4839f41b-ea92-42bf-8ae0-78cdb73b5fd5)

But in this users object there are only keys for firstname, lastname, email and id but not the other custom fields. So I can´t access the avatars url in this way.

Any ideas what would work?

**Comment by SaschaValentin** [comment]
Hello @ConorWebb96 ,

sorry, but this isn´t fully answered. I still don´t know how to access the URL of the avatar field in users table.

The little images have to be different for each user, but they are still the same.

**Reply by ConorWebb96** [reply]
Hey @SaschaValentin,

You can do `userobject[0].firstName` you only really need to use the ["first name"] syntax whenever there is a space in the item you're referencing.

Can you screenshot the fields of the table you're referring to?

Thanks in advance.

**Comment by SaschaValentin** [comment]
Hi @ConorWebb96,

I´m using news table on a MYSQL database:

![ScreenShot 043 Budibase – Mozilla Firefox](https://github.com/Budibase/budibase/assets/48714823/e75f2a16-bb9b-4c4a-a159-604564d6a785)

which refers in column "User" to the Budibase internal user DB. And there I have a field avatar:

![ScreenShot 042 Budibase – Mozilla Firefox](https://github.com/Budibase/budibase/assets/48714823/4bc40abc-6735-45ac-a8d5-884c782c55e0)

So when you look on the upper screenshots you se, that "Sascha Valentin" and "Max Mustermann" have the same avatar - what is wrong.

The user object on the news table only contains firstname, lastname, email and id but not the other custom fields. So I can´t access the avatar.

**Reply by ConorWebb96** [reply]
Hey @SaschaValentin,

The user column only allows for you to pull out the fields which you have listed. Feel free to add a feature request to allow user avatars to be set in a similar way to Firstname Lastname etc. That way you would then be able to also retrieve this image as well.

I hope this makes sense.

**Reply by SaschaValentin** [reply]
Okay, I will do so😉

But until something like this is realized there is no way to achieve my goal? Some workaround?

**Reply by ConorWebb96** [reply, marked as answer]
Hey @SaschaValentin,

You could create a user table in MySQL and use the user column to connect the firstname, lastname etc here. Then possibly use an S3 bucket something along those lines to save an avatar image, reference the url in MySQL table. From there you should have access to everything in the users table from this MySQL table with the additional custom fields. 

I hope this makes sense.
